docs/catalog.py

```diff
@@ -44,10 +69,23 @@ def make_content(artifact, label, all_labels):
         result += "+" * len(explanation_str) + "\n\n"
         result += artifact_class.__doc__ + "\n"
 
+    subtypes = all_subtypes_of_artifact(artifact)
+    subtypes = list(set(subtypes))
+    subtypes.remove(artifact_type)  # this was already documented
+    for subtype in subtypes:
+        subtype_class = Artifact._class_register.get(subtype)
+        subtype_class_name = subtype_class.__name__
+        if subtype_class.__doc__:
+            explanation_str = f"Explanation about `{subtype_class_name}`"
+            result += f"\n{explanation_str}\n"
+            result += "+" * len(explanation_str) + "\n\n"
+            result += subtype_class.__doc__ + "\n"
+
     references = []
     for label in all_labels:
-        if f'"{label}"' in result:
-            references.append(f":ref:`{label} <{label}>`")
+        label_no_catalog = label[8:]  # skip over the prefix 'catalog.'
+        if f'"{label_no_catalog}"' in result:
+            references.append(f":ref:`{label_no_catalog} <{label}>`")
```

Comment:
good catch!

Reply:
Thanks!  perhaps you know how to push the reference into the main json block, in the right positions? 
E.g., I wanted to replace  "tasks.summarization.abstractive" in [this card](https://unitxt.readthedocs.io/en/latest/catalog.cards.xlsum.english.html), with a reference [to here](https://unitxt.readthedocs.io/en/latest/catalog.tasks.summarization.abstractive.html) but I do not manage to get the html renderer to look inside json blocks. 

Do you know how?